Source/WebCore/loader/archive/cf/LegacyWebArchive.cpp

```diff
@@ -616,6 +617,32 @@ RefPtr<LegacyWebArchive> LegacyWebArchive::create(const String& markupString, Lo
 
                 subresources.append(resource.releaseNonNull());
             }
+
+            if (!subresourcesDirectoryName.isNull() && is<HTMLLinkElement>(node)) {
+                auto& element = downcast<HTMLLinkElement>(node.get());
+                auto index = subresources.findIf([&](auto& resource) {
+                    return resource->url() == element.href();
```

Comment:
Not every link element is a stylesheet.
We should probably only do this when sheet() returns a non-null value.
e.g. rel=preload resources should probably be ignored.

Reply:
In this patch we only replace file content when `m_sheet` is non-null, as `HTMLLinkElement::sytleSheetContentWithReplacementURLs` returns null when `m_sheet` is null and we don't overwrite the file when content string is null.

Reply:
What I meant is that we should probably do that check here instead of always calling `styleSheetContentWithReplacementURLs`. By the way, there is a typo in that function name: `sytle`.

Reply:
Oh will correct; and I will add a check here